scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -916,6 +935,29 @@ def apply_filters(self, **kwargs):
                 # handle quarantined tests
                 self.handle_quarantined_tests(instance, plat)
 
+                # Check if scripting is available
+                if self.scripting:
+                    matched_scripting = self.scripting.get_matched_scripting(instance.testsuite.id, plat.name)
+                    if matched_scripting:
+                        # Define a function to validate if the platform is supported by the matched scripting
+                        def validate_boards(platform_scope, platform_from_yaml):
+                            return any(board in platform_scope for board in platform_from_yaml)
+
+                        # Define the types of scripts we are interested in
+                        script_types = {'pre_script': 'pre_script','post_flash_script': 'post_flash_script', 'post_script': 'post_script' }
+
+                        # Iterate over all DUTs to set the appropriate scripts if they match the platform and are supported
+                        for dut in self.env.hwm.duts:
+                            if dut.platform == plat.name and validate_boards(plat.name, matched_scripting.platforms):
+                                for script_type, script_name in script_types.items():
+                                    script = getattr(matched_scripting, script_type, None)
+                                    if script:
+                                        if Path(script).is_file():
+                                            logger.info(f"{script_name} {script} will be executed")  # Validate if test exists
+                                            setattr(dut, script_name, script)
```

Comment:
if the the `dut` already have a script defined, then it will be overridden - this behavior should be documented.

imo, the new scripting schema should have an attribute like `override` (default = true ?) to resolve which configuration takes  precedence.

Reply:
this is explained in twister.rst, you manage this which script will be executed so you don't need adding extra parameter

Reply:
My concern is around the complexity introduced with this feature now with 3 configuration sources: command line, device map, and this new config (not mention its regexp selection rules).

Looks like a DUT can be selected several times matching different test case & platform pattens, so  without explicit constrains its pre-/post- script constellation will, quite likely, composed in a funny way.